libcudacxx/include/cuda/std/__iterator/iter_move.h

```diff
@@ -89,6 +89,7 @@ _CCCL_CONCEPT __just_deref = _CCCL_FRAGMENT(__just_deref_, _Tp);
 
 struct __fn
 {
+  _CCCL_EXEC_CHECK_DISABLE
```

Comment:
Please also add that to the other operators

Reply:
`return *_CUDA_VSTD::forward<_Ip>(__i);` still only calls `operator*` on a unknown iterator